### PR TITLE
Ignore API/SQL enrichments on error

### PR DESCRIPTION
**Pull request by pondzix**
Changes in the iglu central schemas - https://github.com/snowplow/iglu-central/pull/1273

### CODE REVIEW

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala [hunk 1]
```diff
@@ -69,9 +69,16 @@ object ApiRequestEnrichment extends ParseableEnrichment {
           inputs,
           CirceUtils.extract[HttpApi](c, "parameters", "api", "http").toValidatedNel,
           CirceUtils.extract[List[Output]](c, "parameters", "outputs").toValidatedNel,
-          CirceUtils.extract[Cache](c, "parameters", "cache").toValidatedNel
-        ).mapN { (inputs, api, outputs, cache) =>
-          ApiRequestConf(schemaKey, inputs, api, outputs, cache)
+          CirceUtils.extract[Cache](c, "parameters", "cache").toValidatedNel,
+          CirceUtils
+            .extract[Option[Boolean]](c, "parameters", "ignoreOnError")
```
Comment (pondzix) [inline modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala:74]: I guess we would have to also include this new param in a new version (1.0.2?) on [iglucentral](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow.enrichments/api_request_enrichment_config/jsonschema/1-0-1)?

**Review by istreeter — APPROVED** [review]

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala (hunk 1 above)
Comment (benjben) [reply modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala:74]: [Indeed](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow.enrichments/api_request_enrichment_config/jsonschema/1-0-1#L160)

**Comment by benjben** [comment]
Is a timeout also considered as an error ? I think that it should

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala (hunk 1 above)
Comment (benjben) [reply modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala:74]: Have you thought of the roll out / roll back plan ? I wonder if Enrich should be able to parse both versions of the config file until the transition is complete ?

**Review by benjben — APPROVED** [review]

File: modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala (hunk 1 above)
Comment (pondzix) [reply modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/apirequest/ApiRequestEnrichment.scala:74]: `supportedSchema` here is: 

```
override val supportedSchema =
    SchemaCriterion(
      "com.snowplowanalytics.snowplow.enrichments",
      "api_request_enrichment_config",
      "jsonschema",
      1,
      0
    )
```

1.0.0, 1.0.1 and 1.0.2 will work. And we parse `ignoreOnError` as `Option[Boolean]` so it's optional with default value `false`.

**Pull request merged**